crates/ethrpc/src/lib.rs

```diff
@@ -33,6 +33,7 @@ pub type AlloyProvider = DynProvider;
 pub struct Web3<T: Transport = DynTransport> {
     pub legacy: web3::Web3<T>,
     pub alloy: AlloyProvider,
+    pub node_url: String,
```

Comment:
Is it possible to avoid having this string? If we can't avoid it it should be turned into a `Arc<str>` or `Arc<String>` to make cloning it cheap.